cpp/tests/reshape/byte_cast_tests.cpp

```diff
@@ -0,0 +1,322 @@
+     {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xf8, 0x7f},
+     {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xf8, 0x7f},
+     {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xf0, 0x7f},
+     {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xf0, 0xff}});
+
+  auto const output_fp64_flipped = cudf::byte_cast(fp64_col, cudf::flip_endianness::YES);
+  auto const output_fp64         = cudf::byte_cast(fp64_col, cudf::flip_endianness::NO);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output_fp64_flipped->view(), fp64_flipped_expected, true);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output_fp64->view(), fp64_expected, true);
+}
+
+TEST_F(ByteCastTest, fp64ValuesWithNulls)
+{
+  using limits      = std::numeric_limits<double>;
+  auto odd_validity = make_counting_transform_iterator(0, [](auto i) { return i % 2; });
+
+  fixed_width_column_wrapper<double> const fp64_col(
+    {double(0.0), double(100.0), double(-100.0), limits::min(), limits::max()}, {0, 1, 0, 1, 0});
+  // auto fp64_data = fixed_width_column_wrapper<uint8_t>{
+  //   0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff, 0x40, 0x59, 0x00, 0x00, 0x00, 0x00,
+  //   0x00, 0x00, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff, 0x00, 0x10, 0x00, 0x00,
+  //   0x00, 0x00, 0x00, 0x00, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+  auto fp64_data = fixed_width_column_wrapper<uint8_t>{
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40, 0x59, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0xc0, 0x59, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x7f, 0xef, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
+  auto fp64_expected = cudf::make_lists_column(
+    5,
+    std::move(fixed_width_column_wrapper<cudf::size_type>{0, 8, 16, 24, 32, 40}.release()),
+    std::move(fp64_data.release()),
+    3,
+    detail::make_null_mask(odd_validity, odd_validity + 5));
+
+  auto const output_fp64 = cudf::byte_cast(fp64_col, cudf::flip_endianness::YES);
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(output_fp64->view(), fp64_expected->view());
+}
+
+TEST_F(ByteCastTest, StringValues)
+{
+  strings_column_wrapper const strings_col(
+    {"", "The quick", " brown fox...", "!\"#$%&\'()*+,-./", "0123456789:;<=>?@", "[\\]^_`{|}~"});
+  lists_column_wrapper<int8_t> const strings_expected(
+    {{},
+     {0x54, 0x68, 0x65, 0x20, 0x71, 0x75, 0x69, 0x63, 0x6b},
+     {0x20, 0x62, 0x72, 0x6f, 0x77, 0x6e, 0x20, 0x66, 0x6f, 0x78, 0x2e, 0x2e, 0x2e},
+     {0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29, 0x2a, 0x2b, 0x2c, 0x2d, 0x2e, 0x2f},
+     {0x30,
+      0x31,
+      0x32,
+      0x33,
+      0x34,
+      0x35,
+      0x36,
+      0x37,
+      0x38,
+      0x39,
+      0x3a,
+      0x3b,
+      0x3c,
+      0x3d,
+      0x3e,
+      0x3f,
+      0x40},
+     {0x5b, 0x5c, 0x5d, 0x5e, 0x5f, 0x60, 0x7b, 0x7c, 0x7d, 0x7e}});
+
+  auto const output_strings = cudf::byte_cast(strings_col, cudf::flip_endianness::YES);
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output_strings->view(), strings_expected, true);
+}
```

Comment:
I suggest adding some tests with columns that have been split or sliced (i.e. have a non-zero offset).